src/client/utils/download/download.ts

```diff
@@ -38,12 +38,24 @@ export function getMIMEType(fileType: string) {
   }
 }
 
-export function download({ dataset, options }: DataSetWithTabOptions, fileFormat: FileFormat, fileName?: string): void {
-  const type = `${getMIMEType(fileFormat)};charset=utf-8`;
-  const blob = new Blob([datasetToFileString(dataset, fileFormat, options)], { type });
-  if (!fileName) fileName = `${new Date()}-data`;
-  fileName += `.${fileFormat}`;
-  filesaver.saveAs(blob, fileName, true); // true == disable auto BOM
+export function download({ dataset, options }: DataSetWithTabOptions, fileFormat: FileFormat, fileName?: string, encoding = "utf-8"): void {
+  function save(part: string | Buffer) {
+    const type = `${getMIMEType(fileFormat)};charset=${encoding}`;
+    const blob = new Blob([part], { type });
+    if (!fileName) fileName = `${new Date()}-data`;
+    fileName += `.${fileFormat}`;
+    filesaver.saveAs(blob, fileName, true); // true == disable auto BOM
+  }
+
+  const result = datasetToFileString(dataset, fileFormat, options);
+  encoding = encoding === "" ? "utf-8" : encoding;
```

Comment:
There's already a default `utf-8` in function declaration, this line can be removed. `""` is just an invalid value like `"foobar"`, we shouldn't treat it differently.

Reply:
cool

Reply:
resolved, thanks